README.md
# litRiddle: R package for exploring literary quality

**Authors:** Maciej Eder, Karina van Dalen-Oskam, Joris van Zundert, Saskia Lensink<br/>
**License:** [GPL-3](https://opensource.org/licenses/GPL-3.0)


This is an R package being a part of the Riddle of Literary Quality project.

The package contains the data of a reader survey about fiction in Dutch, a description of the novels the readers rated, and the results of stylistic measurements of the novels. The package also contains functions to combine, analyze, and visualize these data.

See: [https://literaryquality.huygens.knaw.nl/](https://literaryquality.huygens.knaw.nl/) for further details. Information in Dutch about the package can be found at [https://karinavdo.github.io/RaadselLiteratuur/02_07_data_en_R_package.html](https://karinavdo.github.io/RaadselLiteratuur/02_07_data_en_R_package.html).

If you use `litRiddle` in your academic publications, please consider citing the following references: 

Maciej Eder, Lensink, S., Van Zundert, J.J., and Van Dalen-Oskam, K.H. (2022). “Replicating The Riddle of Literary Quality: The LitRiddle Package for R.” In _Digital Humanities 2022 Conference Abstracts_, 636–637. Tokyo: The University of Tokyo / DH2022 Local Organizing Committee. https://dh2022.dhii.asia/abstracts/163.

Karina van Dalen-Oskam (2023). _The Riddle of Literary Quality: A Computational Approach._ Amsterdam University Press.

## Installation

Install the package from the CRAN repository:

``` R
install.packages("litRiddle")
```

Alternatively, try installing it directly from the current GitHub repository:

``` R
library(devtools)
install_github("karinavdo/LitRiddleData", build_vignettes = TRUE)
```


## Usage

The package is complemented with a concise vignette, which can be accessed via:

``` R
vignette("litRiddle")
```
This document will briefly guide the users through the functions of the package, but please expect some rough edges: the document will be revised in the future. A comprehensive yet rather low-level documentation of the package is provided – obviously – via the generic `help()` function in R. Try typing e.g.:


``` R
help(books)
help(respondents)
help(reviews)
help(frequencies)
help(motivations)
help(motivations.text)
help(combine.all)
help(explain)
help(find.dataset)
help(get.columns)
help(make.table)
help(make.table2)
help(order.responses)
help(litRiddle) # for the general description of the package
```

Each function of the package, as well as each dataset, is supplemented by a help page.

A concise description of the package's functionalities (in Dutch) is also provided here: 
[https://karinavdo.github.io/RaadselLiteratuur/02_07_data_en_R_package.html](https://karinavdo.github.io/RaadselLiteratuur/02_07_data_en_R_package.html).


## General info

**Karina van Dalen-Oskam** (2023). _The Riddle of Literary Quality: A Computational Approach._ Amsterdam University Press.

**Karina van Dalen-Oskam** (2021). _Het raadsel literatuur. Is literaire kwaliteit meetbaar?_ Amsterdam University Press.

**Maciej Eder, Saskia Lensink, Joris van Zundert, Karina van Dalen-Oskam** (2022). “Replicating The Riddle of Literary Quality: The litRiddle package for R”, in: _Digital Humanities 2022 Conference Abstracts._ The University of Tokyo, Japan, 25–29 July 2022, p. 636-637 https://dh2022.dhii.asia/dh2022bookofabsts.pdf

**Corina Koolen, Karina van Dalen-Oskam, Andreas van Cranenburgh, Erica Nagelhout** (2020). Literary quality in the eye of the Dutch reader: The National Reader Survey. _Poetics_ 79: 101439, https://doi.org/10.1016/j.poetic.2020.101439.


More publications can be found at [https://literaryquality.huygens.knaw.nl/?page_id=588](https://literaryquality.huygens.knaw.nl/?page_id=588)




NEWS.md
# LitRiddle 1.0.0:9001

2023-06-16 Joris J. van Zundert <joris.van.zundert@gmail.com>

## Changes breaking backwards compatibility

Data frame `books`: updated column order, naming, and types.

## Additions/changes

Added column `riddle.code.english` to data frame `books`.  Updated README, NEWS, vignette, manuals. 
Bumped development version indicator 1 up.


# LitRiddle 1.0.0:9000

2023-06-02 Joris J. van Zundert <joris.van.zundert@gmail.com>

## Changes breaking backwards compatibility

* Removed the column "motivations" from data table "reviews" to avoid putting XML or JSON etc. in data cells

## Additions

* Added a data table "motivations" (`data/motivations.rda`) that has columns "motivation.id", "book.id", "respondent.id", "paragraph.id", "sentence.id", "token", "lemma", and "upos". It is a single table solution to storing all tokens of all motivations, one token per row/line.
* Added `R/motivations.R` module/script to provide convenience functions transforming the data table "motivations" into tables having one sentence of each motivation per row, or having the full plain text of each motivation per row.
* Added all necessary and related documentation to explain changes and usage.
* Updated `DESCRIPTION.md`, `NAMESPACE.md`, `R/columnNames.R`, `R/explain.R`.

* Added a `NEWS.md` file to track changes to the package.


R/zzz.R

.onAttach <- function(lib, pkg) {
    packageStartupMessage("litRiddle version: ", utils::packageVersion("litRiddle"), 
        "\n", 
        "\nThank you for working with the litRiddle package. We would greatly",
        "\nappreciate properly citing this software if you find it of use.",
        "\nWhen citing, you can refer to this software as:", 
        "\n    Eder, M., Van Zundert, J., Lensink, S., and Van Dalen-Oskam, K. (2021). ", 
        "\n    litRiddle: Dataset and Tools to Research the Riddle of Literary Quality ",
        "\n    (1.0.0). CRAN. <https://cran.r-project.org/web/packages/litRiddle/index.html>", 
        "\nIf you prefer to cite a publication, you can cite:",
        "\n    Maciej Eder, Lensink, S., Van Zundert, J.J., and Van Dalen-Oskam, K.H. ",
        "\n    (2022). “Replicating The Riddle of Literary Quality: The LitRiddle ",
        "\n    Package for R.” In _Digital Humanities 2022 Conference Abstracts_, ",
        "\n    636–637. Tokyo: The University of Tokyo / DH2022 Local Organizing ",
        "\n    Committee. <https://dh2022.dhii.asia/abstracts/163>.",
        "\n",
        "\nTo get full BibTeX entry, type: citation(\"litRiddle\").",
        "\n"
    )
}


R/makeTable.R
# -------------------------------------- # 
#  	MAKE TABLE 	                         # 
# -------------------------------------- #

# make a table of frequency counts for one variable, and plot a histogram of the results.

make.table <- function(table.of = ..., 
                       plot = TRUE, 
                       xlab = table.of, 
                       ylab = "count", 
                       title = table.of,
                       barcolor = "grey", 
                       barfill = "darkgrey"){
#
    # a hack to silence package warnings  
    data("books", package = "litRiddle", envir = environment()) 
    books <- get("books", envir = environment()) 
    #
    data("respondents", package = "litRiddle", envir = environment()) 
    respondents <- get("respondents", envir = environment()) 
    #
    data("reviews", package = "litRiddle", envir = environment()) 
    reviews <- get("reviews", envir = environment()) 
    #



    if (table.of %in% colnames(reviews)){
        dat.x = as.data.frame(reviews[,table.of])
        table.x = table(reviews[,table.of])
    } else if (table.of %in% colnames(respondents)) {
        dat.x = as.data.frame(respondents[,table.of])
        table.x = table(respondents[,table.of])
    } else if (table.of %in% colnames(books)){
        dat.x = as.data.frame(books[,table.of])
        table.x = table(books[,table.of])
    } else {
        message("Please provide a valid column name. The column names can be printed on the screen by typing 'print.columns()' in the console.")
    }
  
    # some visuals
    if(exists('table.x') & plot == TRUE){
    
        g = ggplot(dat.x, aes(x=dat.x[,1]))
        g = g + geom_bar(colour=barcolor, fill=barfill)
        g = g + xlab(xlab) + ylab(ylab) + ggtitle(title)
        plot(g)
    
        return(table.x)
    
    } else if(exists('table.x')){
        return(table.x)
    } 
}



R/makeTable2.R
# --------------------------------------------------- #
# 	MAKE TABLE OF X SPLIT BY Y		      #
# --------------------------------------------------- #

# Makes a table of a variable x, and will split this table by variable y.


make.table2 = function(table.of = ..., 
                       split = ..., 
                       plot = TRUE, 
                       xlab = table.of, 
                       ylab = "counts",
                       barcolor = "grey", 
                       barfill = "darkgrey"){
  

    # a hack to silence package warnings  
    data("books", package = "litRiddle", envir = environment()) 
    books <- get("books", envir = environment()) 
    #
    data("respondents", package = "litRiddle", envir = environment()) 
    respondents <- get("respondents", envir = environment()) 
    #
    data("reviews", package = "litRiddle", envir = environment()) 
    reviews <- get("reviews", envir = environment()) 
    #

    # combine all to make sure arguments match in numbers of rows
    dat = combine.all()
    dat = as.data.frame(dat)  # to make referring to variable column names possible
  
    # if the split variable has more than 31 unique values (number of unique week numbers present in the dataset), 
    # do not continue:
    if (length(unique(dat[,split])) > 31) {
        message("The 'split-by' variable has many unique values, which will ",
                "make the output \nvery hard to process. Please provide",
                "a 'split-by' variable that contains \nless unique values.")
    } else {
        col1 = dat[,table.of]
        col2 = dat[,split]
        dat = data.frame(col1, col2)
        table2 = table(dat[, 2], dat[, 1])
    
        # some visuals
        if(exists('table2') & plot == TRUE){
      
            g = ggplot(dat, aes(x=dat[,1]))
            g = g + facet_wrap(~dat[,2], scales = "free_x")
            g = g + geom_bar(colour=barcolor, fill=barfill)
            g = g + xlab(xlab) + ylab(ylab) 
            plot(g)
            
            # how to return two things?
            return(table2)
      
        } else if(exists('table2')){
            return(table2)
        }
    }
}




R/columnNames.R
## ------------------------------------------------------- ##
##           print column names
## ------------------------------------------------------- ##

# function that creates a list of all the column names from all three datasets
print.columns <- function( ){
    # a hack to silence package warnings  
    data("books", package = "litRiddle", envir = environment()) 
    books <- get("books", envir = environment()) 
    #
    data("respondents", package = "litRiddle", envir = environment()) 
    respondents <- get("respondents", envir = environment()) 
    #
    data("reviews", package = "litRiddle", envir = environment()) 
    reviews <- get("reviews", envir = environment()) 
    #
    
    colnames.books = colnames(books)
    colnames.resp = colnames(respondents)
    colnames.reviews = colnames(reviews)
    columns = list(books = colnames.books, respondents = colnames.resp, reviews = colnames.reviews)
    print(columns)
}



R/findDataset.R
# ---------------------------------------- # 
#       FIND DATASET                       #
# ---------------------------------------- # 


# Return the name of the dataset where a column can be found.

find.dataset = function(name = ...){
    
    # a hack to silence package warnings  
    data("books", package = "litRiddle", envir = environment()) 
    books <- get("books", envir = environment()) 
    #
    data("respondents", package = "litRiddle", envir = environment()) 
    respondents <- get("respondents", envir = environment()) 
    #
    data("reviews", package = "litRiddle", envir = environment()) 
    reviews <- get("reviews", envir = environment()) 
    #

    if(name == "book.id"){
        print(c("books", "reviews"))
    } else if(name == "respondent.id"){
        print(c("respondents", "reviews"))
    } else if(name %in% colnames(books)){
        print("books")
    } else if(name %in% colnames(respondents)){
        print("respondents")
    } else if(name %in% colnames(reviews)){
        print("reviews")
    } else {
        stop("This is not a valid column name. Please type <print.columns()> to see which column names exist.")
    }

}



R/orderResponses.R
# -------------------------------------- # 
#  	ORDER RESPONSES			             # 
# -------------------------------------- #

# function that transforms the survey responses into ordered factors


order.responses <- function(bookratings.or.readingbehavior = ...){

    # a hack to silence package warnings  
    data("books", package = "litRiddle", envir = environment()) 
    books <- get("books", envir = environment()) 
    #
    data("respondents", package = "litRiddle", envir = environment()) 
    respondents <- get("respondents", envir = environment()) 
    #
    data("reviews", package = "litRiddle", envir = environment()) 
    reviews <- get("reviews", envir = environment()) 
    #
    
    
    if(bookratings.or.readingbehavior == "bookratings"){
        dat = reviews
        cols1 = c(3,5)
        cols2 = c(4,6)
        dat[,cols1] = as.data.frame(lapply(dat[,cols1], function(X){ordered(X,
            levels = 1:7, 
            labels = c("very bad", "bad", "a bit bad", 
            "neutral", "a bit good", "good", 
        "very good"))}))
        dat[,cols2] = as.data.frame(lapply(dat[,cols2], function(X){ordered(X,
            levels = 1:7, 
            labels = c("absolutely not literary", "non-literary", 
            "not very literary", "between literary and non-literary",
        "a bit literary", "literary", "very literary"))}))
        
    } else if(bookratings.or.readingbehavior == "readingbehavior"){
        dat = respondents
        cols = 8:24
        dat[,cols] = as.data.frame(lapply(dat[,cols], function(X){ordered(X,
            levels = 1:5, 
            labels = c("completely disagree", "disagree", 
        "neutral", "agree", "completely agree"))}))
        
    } else {
        stop("Please type in either <'bookratings'> or <'readingbehavior'>")
    }
    
    dat = as_tibble(dat)
    return(dat)
    
}


# levels "quality.read" and "quality.notread": "very bad", "bad", "a bit bad", "neutral", "a bit good", "good", "very good", NA.
# levels "literariness.read" and "literariness.notread": "absolutely not literary", "non-literary", "not very literary", 
#         "between literary and non-literary","a bit literary", "literary", "very literary", NA.

# levels statements 4 / 12: "completely disagree", "disagree", "neutral", "agree", "completely agree", "NA"




R/manual_draft_litRiddle.R


dummy.function = function() {


## ------------------------------------------------------- ##
##           print column names
## ------------------------------------------------------- ##

# function that creates a list of all the column names from all three datasets


## Usage

# Use this function to print the names of all available columns
# Don't forget the parentheses at the end of the function!
print.columns()








# ----------------------------------------------- # 
#         COMBINE ALL                             #
# ----------------------------------------------- #

# Function to combine all information of the survey, reviews, and books into one big dataframe
# The user can specify whether or not s/he want to also load the freqTable with the frequency
# counts of the word n-grams of the books. Default = FALSE. 


## Usage 

# combine and load all data from the books, respondents and reviews into 
# a new dataframe (tibble format)
dat = combine.all(load.freq.table=FALSE)

# combine and load all data from the books, respondents and reviews into 
# a new dataframe (tibble format), and additionally also load the frequency
# table of all word 1grams of the corpus used. 
dat = combine.all(load.freq.table=TRUE)








# ------------------------------------- #
#          explain variables            #
# ------------------------------------- #


# Function that lists an short explanation of what the different column names refer to
# and what their levels consist of.


## Usage
explain('books')
explain('reviews')
explain('respondents')








# ---------------------------------------- # 
#       FIND DATASET                       #
# ---------------------------------------- # 


# Return the name of the dataset where a column can be found.


## Usage
find.dataset('book.id')
find.dataset('age.resp')
find.dataset('wrong.column.name')







# -------------------------------------- # 
#  	MAKE TABLE 	                         # 
# -------------------------------------- #

# make a table of frequency counts for one variable, and plot a histogram of the results.


## Usage 

# print column names to see which variables you can choose from:
print.columns()

# don't forget to put parentheses around the variable name!
make.table(table.of = 'age.resp')

# You can also adjust the x label, y label, title, and colors. Don't forget to use parentheses!
make.table(table.of = 'age.resp', xlab = 'age respondent', ylab = 'number of people', 
           title = 'Distribution of respondent age', barcolor = 'red', barfill = 'white')


           






# --------------------------------------------------- #
# 	MAKE TABLE OF X SPLIT BY Y		      #
# --------------------------------------------------- #

# Makes a table of a variable x, and will split this table by variable y.



## Usage

# print column names to see which variables you can choose from:
print.columns()

# don't forget to put parentheses around the variable name!
make.table2(table.of = 'age.resp', split = 'gender.resp')
make.table2(table.of = 'literariness.read', split = 'gender.author')

# Note that you can only provide an argument to the 'split' variable that has less than 31 unique 
# values, to avoid uninterpretable outputs:
make.table2(table.of = 'age.resp', split = 'zipcode') 

# You can also adjust the x label, y label, title, and colors. Don't forget to use parentheses!
make.table2(table.of = 'age.resp', split = 'gender.resp', xlab = 'age respondent', ylab = 'number of people', 
            barcolor = 'purple', barfill = 'yellow')
make.table2(table.of = 'literariness.read', split = 'gender.author', xlab = 'Overall literariness scores', 
            ylab = 'number of people', barcolor = 'black', barfill = 'darkred')








# -------------------------------------- # 
#  	ORDER RESPONSES			                 # 
# -------------------------------------- #

# function that transforms the survey responses into ordered factors


# levels "quality.read" and "quality.notread": "very bad", "bad", "a bit bad", "neutral", "a bit good", "good", "very good", NA.
# levels "literariness.read" and "literariness.notread": "absolutely not literary", "non-literary", "not very literary", 
#         "between literary and non-literary","a bit literary", "literary", "very literary", NA.

# levels statements 4 / 12: "completely disagree", "disagree", "neutral", "agree", "completely agree", "NA"


## Usage

# Use either 'bookratings' or 'readingbehavior' to specify which of the survey questions needs to be 
# changed into ordered factors. Don't forget to use parentheses!

# to create a data frame with ordered factor levels of the questions on reading behavior
dat.reviews = order.responses('readingbehavior')
str(dat.reviews)

# to create a data frame with ordered factor levels of the book ratings:
dat.ratings = order.responses('bookratings')
str(dat.ratings)


# likert excluded so far
# see here: https://github.com/jbryer/likert






}






R/explain.R
# ------------------------------------- #
#          explain variables            #
# ------------------------------------- #


# Function that lists an short explanation of what the different column names refer to
# and what their levels consist of.

explain <- function(dataset = ""){
    
  if(dataset == "books"){
    cat("The 'books' dataset contains information on several details of the 401 
different books used in the survey.
        
Here follows a list with the different column names and an explanation of
the information they contain:

1. short.title                A short name containing the author's name and 
                              (a part of) the title;
2. author                     Last name and first name of the author;
3. title                      Full title of the book;
4. title.english              Full title of the book in English;
5. genre                      Genre of the book. 
                              There are four different genres:
                              a) Fiction; b) Romantic; c) Suspense; d) Other;
6. book.id                    Unique number to identify each book;
7. riddle.code                More complete list of genres of the books. 
                              Contains 13 categories --- to see which, type
                              `levels(books$riddle.code)` in the terminal;
8. riddle.code.english        Translation of code in column 7 in English;
9. translated                 'yes' if the book has been translated,
                              'no' if not;
10. gender.author              The gender of the author: 
                              Female, Male, Unknown/Multiple;
11. origin.author             The country of origin of the author. 
                              Note that short country codes have been used 
                              instead of the full country names;
12. original.language         The original language of the book. Note that short
                              language codes have been used, instead of the full
                              language names;
13. inclusion.criterion       In what category a book has been placed, either
                              a) bestseller; b) boekenweekgeschenk; 
                              c) library; or d) literair juweeltje;
14. publication.date          Publication date of the book, YYYY-MM-DD format;
15. first.print               Year in which the first print appeared;
16. publisher                 Publishers of the books;
17. word.count                Word count, or total number of words (tokens)
                              used in a book;
18. type.count                Total number of unique words (types) used in book;
19. sentence.length.mean      Average sentence length in a book (in words);
20. sentence.length.variance  Standard deviation of the sentence length;
21. paragraph.count           Total number of paragraphs in a book;
22. sentence.count            Total number of sentences in a book;
23. paragraph.length.mean     Average paragraph length in a book (in words); 
24. raw.TTR                   Lexical diversity, or type-token ratio, which 
                              gives an indication of how diverse the word use 
                              in a book is;
25. sampled.TTR               Unlike the raw type-token ratio, the sampled 
                              TTR is significantly more resistant to text 
                              size, and thus it should be preferred over the 
                              raw TTR. 
\n")
    
  } else if(dataset == "reviews"){
    
    cat("The 'reviews' dataset contains four different ratings that were given 
to 401 different books.
        
Here follows a list with the different column names and an explanation of
what information they contain:

1. respondent.id          Unique number for each respondent of the survey;
2. book.id                Unique number to identify each book;
3. quality.read           Rating on the quality of a book that a respondent
                          has read. Scale from 1 - 7, with 1 meaning 
                          'very bad' and 7 meaning 'very good';
4. literariness.read      Rating on how literary a respondent found a book
                          that s/he has read. Scale from 1 - 7, with 1 meaning 
                          'not literary at all' and 7 meaning 'very literary';
5. quality.notread        Rating on the quality of a book that a respondent
                          has not read. Scale from 1 - 7, with 1 meaning 
                          'very bad' and 7 meaning 'very good';
6. literariness.notread   Rating on how literary a respondent found a book that
                          s/he has not read. Scale from 1 - 7, with 1 meaning 
                          'not literary at all' and 7 meaning 'very literary';
7. book.read              1 or 0: 1 indicates that the respondent read 
                          the book, 0 indicates the respondent did not 
                          read the book but had an opinion about 
                          the literary quality of the book.
\n")
    
  } else if(dataset == "respondents"){
    
    cat("The 'respondents' dataset contains information on the people that participated 
in the survey.
        
Here follows a list with the different column names and an explanation of
what information they contain:

1. respondent.id      Unique number for each respondent of the survey;
2. gender.resp        Gender of the respondent: Female, Male, NA;
3. age.resp           Age of the respondent;
4. zipcode            Zip code of the respondent;
5. education          Education level, containing 8 levels (see which
                      levels by typing 'levels(respondents$education)'
                      in the terminal);
6. books.per.year     Number of books read per year by each respondent;
7. typically.reads    Typical genre of books that a respondent reads, 
                      with three levels a) Fiction; b) Non-fiction;
                      c) both;
8. how.literary       Answer to the question 'How literary a reader do 
                      you consider yourself to be?', where respondents
                      could fill in a number from 1 - 7, with 1 meaning
                      'not literary at all' and 7 meaning 'very literary';
9. s.4a1              Answer to the question: 'I like reading novels that 
                      I can relate to my own life'. Scale from 1 - 5, with 
                      1 meaning 'completely disagree', and 5 meaning 
                      'completely agree';
10. s.4a2             Answer to the question: 'The story of a novel is what 
                      matters most to me'. Scale from 1 - 5; 
11. s.4a3             Answer to the question: 'The writing style in a book 
                      is important to me'. Scale from 1 - 5;
12. s.4a4             Answer to the question: 'I like searching for deeper 
                      layers in a novel'. Scale from 1 - 5;
13. s.4a5             Answer to the question: 'I like reading literature'. 
                      Scale from 1 - 5;
14. s.4a6             Answer to the question: 'I read novels to discover new 
                      worlds and unknown time periods'. Scale from 1 - 5;
15. s.4a7             Answer to the question: 'I mostly read novels during my 
                      vacation'. Scale from 1 - 5;
16. s.4a8             Answer to the question: 'I usually read several novels at 
                      the same time'. Scale from 1 - 5;
17. s.12b1            Answer to the question: 'I like novels based on real 
                      events'. Scale from 1 - 5;
18. s.12b2            Answer to the question: 'I like thinking about a novel's 
                      structure'. Scale from 1 - 5;
19. s.12b3            Answer to the question: 'The writing style in a novel 
                      is of more importance to me than its story'. 
                      Scale from 1 - 5;  
20. s.12b4            Answer to the question: 'I like to get carried away by 
                      a novel'. Scale from 1 - 5;
21. s.12b5            Answer to the question: 'I like to pick my books from 
                      the top 10 list of best sold books'. Scale from 1 - 5;
22. s.12b6            Answer to the question: 'I read novels to be challenged 
                      intellectually'. Scale from 1 - 5;
23. s.12b7            Answer to the question: 'I love novels that are easy 
                      to read'. Scale from 1 - 5;
24. s.12b8            Answer to the question: 'In the evening, I prefer 
                      to read books over watching TV'. Scale from 1 - 5;
25. remarks.survey    Any additional remarks that respondents filled in
                      at the end of the survey;
26. date.time         Date and time of the moment a respondent filled in
                      the survey, format in YYYY-MM-DD HH:MM:SS;
27. week.nr           Number of week in which the respondent filled in 
                      the survey;
28. day               Day of the week in which the respondent filled in
                      the survey.
\n")
   
  } else if(dataset == "motivations"){
    
    cat("The 'motivations' dataset contains all motivations that people provided 
about why they gave a certain book a specific rating. The motivations have been
parsed to provide POS tag information
        
Here follows a list with the different column names and an explanation of
what information they contain:

1. motivation.id      Unique number for each motivation given;
2. respondent.id      Unique number for each respondent;
3. book.id            Unique number of the book the motivation pertains to;
4. paragraph.id       Number of paragraph within the motivation;
5. sentence.id        Number of sentence within the paragraph;
6. token              Token (in sentence order);
7. lemma              Lemma of token;
8. upos               POS tag of token;
\n")

  } else if(dataset == "frequencies"){
    
    cat("This is a dataframe containing numerical values for word frequencies
of the 5000 most frequent words (in a descending order of frequency)
of 401 literary novels in Dutch. The table contains relative frequencies,
meaning that the original word occurrences from a book were divided 
by the total number of words of the book in question.

The row names coincide with the column short.title from the data frame books.
The column names list the 5000 most frequent words in the corpus. 
\n")

  } else {
    
    stop("Please specify either 'books', 'respondents', 'reviews', or 'motivations', \n e.g. explain(\"books\").")
    
  }

  
}

